src/main/java/org/apache/maven/plugins/checkstyle/Violation.java

```diff
@@ -0,0 +1,224 @@
+ */
+public class Violation
+{
+
+  /**
+   * Indicates that a column is not set.
+   */
+  protected static final String NO_COLUMN = "-1";
+
+  /** The source file name relative to the project's root. */
+  private final String source;
+
+  /** File is the absolute file name to the checked file. */
+  private final String file;
+
+  private final String line;
+
+  private String column = NO_COLUMN;
+
+  private final String severity;
+
+  private final String message;
+
+  private final String ruleName;
+
+  private final String category;
+
+  // Leaving out column, because there is no CHECKSTYLE:OFF support.
+
+  /**
+   * Creates a violation instance without a column set.
+   *
+   * @param source
+   *     the source file name, relative to the project's root.
+   * @param file
+   *     the absolute file name in which the violation occurred.
+   * @param line
+   *     the line in the file on which the violation occurred.
+   * @param severity
+   *     the severity of the violation.
+   * @param message
+   *     the message from checkstyle for this violation.
+   * @param ruleName
+   *     the rule name from which this violation was created.
+   * @param category
+   *     the category of the checkstyle violation.
+   */
+  public Violation( String source,
+                    String file,
+                    String line,
+                    String severity,
+                    String message,
+                    String ruleName,
+                    String category )
+  {
+    this.source = Objects.requireNonNull( source );
+    this.file = file;
+    this.line = line;
+    this.severity = Objects.requireNonNull( severity );
+    this.message = Objects.requireNonNull( message );
+    this.ruleName = Objects.requireNonNull( ruleName );
+    this.category = Objects.requireNonNull( category );
+  }
+
+  /**
+   * Returns the source file name relative to the project's root.
+   *
+   * @return the source file name relative to the project's root.
+   */
+  protected String getSource( )
+  {
+    return source;
+  }
+
+  /**
+   * Returns the absolute file name to the checked file.
+   *
+   * @return the absolute file name to the checked file.
+   */
+  protected String getFile( )
+  {
+    return file;
+  }
+
+  /**
+   * Returns the line of in the checked file on which the violation occurred.
+   *
+   * @return the line of in the checked file on which the violation occurred.
+   */
+  protected String getLine( )
+  {
+    return line;
+  }
+
+  /**
+   * Returns the column in which the violation occurred, if available.
+   *
+   * @return the column in which the violation occurred, if available. Otherwise returns {@code "-1"}.
+   */
+  protected String getColumn( )
+  {
+    return column;
+  }
+
+  protected void setColumn( /* Nullable */ String column )
+  {
+    if ( null == column || column.length() < 1 )
+    {
+      this.column = NO_COLUMN;
+      return;
```

Comment:
use else and no return.

Reply:
This would add in complexity (cyclomatic complexity) and would put the main code below into a new indentation level. I would always look for the main code at the first indentation level. I have not seen any written rule to use else. Using else you will get complex code quickly. Is this personal taste or is this written down somewhere?

Reply:
Please change this yourself.

The requested change is purely esthetic and will lead to more nesting, more exit points from that method and more complicated future changes.

If you do not want the main logic to be on the first indentation level of a method, please change it yourself.

Also, please write those rules down. I do not see any documentation about this. This is somewhat deterrent to me as a (almost) first time committer.

Reply:
Done, the new code now looks like existing code. I think this is more or less personal preference, as there is probably no bytecode change and readability depends on routine and how used to are to this style.